### PR TITLE
Fix #30462, out of bounds access in maximum/minimum for certain-sized arrays

### LINKED ISSUE #30462: BoundsError on master but works fine on 1.0.3 (and previous)
A test with nightly builds of GMT.jl package started to fail since about two days ago (18 December). All tests pass with version 1.0.3 but one of them now fails in master. See
https://travis-ci.org/GenericMappingTools/GMT.jl/builds/470272490?utm_source=github_status&utm_medium=notification
The [failure ](https://travis-ci.org/GenericMappingTools/GMT.jl/jobs/470272492#L3122) is about accessing an array out of bounds but the same code has been working with no problem till 2 days ago and still works fine with 1.0.3
```
julia> versioninfo()
Julia Version 1.0.3
Commit 099e826241 (2018-12-18 01:34 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, haswell)
Environment:
  JULIA_PKGDIR = c:\j\.julia
```

### PR DESCRIPTION
**Pull request by jw3126**
#30462 This is still fast:
```julia
julia> using BenchmarkTools; arr = randn(10^6); @benchmark maximum($arr)
[ Info: Recompiling stale cache file /home/jan/.julia/compiled/v1.2/BenchmarkTools/ZXPQo.ji for BenchmarkTools [6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf]
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     557.330 μs (0.00% GC)
  median time:      587.258 μs (0.00% GC)
  mean time:        588.535 μs (0.00% GC)
  maximum time:     855.520 μs (0.00% GC)
  --------------
  samples:          8403
  evals/sample:     1
```

### CODE REVIEW

File: test/reduce.jl [hunk 1]
```diff
@@ -244,6 +244,10 @@ end
     end
 end
 
+# 30462
+# test that there is no out of bounds access
+maximum(randn(128,128))
```
Comment (jw3126) [inline test/reduce.jl]: I would like to add more/better tests that there are no out of bounds accesses. Is there a recommended way to do this? Just make a function call and rely on `--check-bounds=yes`?

Comment (mbauman) [reply test/reduce.jl]: You could also test `maximum(fill(-Inf, 128, 128)) === -Inf`.  It's unlikely that uninitialized memory will be a bitpattern that represents `-Inf`.  But we do run all our tests with `--check-bounds=yes` (excepting in the `test/boundscheck*.jl` files)

**Comment by mbauman** [comment]
Applying same backport tag as the original PR (https://github.com/JuliaLang/julia/pull/30320).

**Comment by StefanKarpinski** [comment]
Bump. This seems good to go doesn't it?
